spec/models/phone_number_spec.rb

```diff
@@ -4,55 +4,134 @@
 
 RSpec.describe PhoneNumber, type: :model do
   describe 'validations' do
-    let(:user) do
-      User.create(username: 'Mike',
-                  biography: 'I am a developer',
-                  age: 25)
+    subject do
+      PhoneNumber.new(phone_number: '123',
+                      label: 0,
+                      main: true,
+                      contact: contact)
     end
 
-    let(:contact) do
-      Contact.create(name: 'Rudolph',
-                     user_id: user.id)
+    let(:user) do
+      User.create!(username: 'Mike',
+                   biography: 'I am a developer',
+                   age: 25)
     end
 
-    subject do
-      PhoneNumber.new(phone_number: '00123456789',
-                      label: 'work',
-                      primary: FalseClass,
-                      contact_id: contact.id)
+    let(:contact) do
+      Contact.create!(name: 'Rudolph',
+                      user: user)
     end
 
-    context 'when all attributes are present' do
+    context 'when needed attributes are present' do
       it 'is valid' do
         expect(subject.valid?).to be_truthy
       end
     end
 
     context 'when number is not present' do
+      before { subject.phone_number = nil }
+
       it 'is not valid without a phone_number' do
```

Comment:
Dá uma revisada nas mensagens do seu teste.
Por exemplo: context 'when something', it 'what happens?' 

Pode dar uma olhada aqui, como escrever mensagens melhores: https://www.betterspecs.org/